sklearn/linear_model/ridge.py

```diff
@@ -469,7 +471,13 @@ def __init__(self, alpha=1.0, fit_intercept=True, normalize=False,
         self.random_state = random_state
 
     def fit(self, X, y, sample_weight=None):
-        X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float64,
+
+        if self.solver in ['svd', 'sparse_cg', 'cholesky', 'lsqr']:
+            _dtype = [np.float64, np.float32]
+        else:
+            _dtype = np.float64
```

Comment:
I think this comment should be in #8769, so that everyone can make sure that the default policy is to support both dtypes.
I'll post it there.